src/fty_discovery_server.cc

```diff
@@ -842,17 +665,61 @@ s_handle_mailbox(fty_discovery_server_t* self, zmsg_t *msg, zpoller_t *poller) {
                 zstr_free(&self->range_scan_config.range);
                 zstr_free(&self->range_scan_config.range_dest);
 
-                if (self->configuration_scan.type == TYPE_LOCALSCAN) {
-                    //Launch localScan
-                    configure_local_scan(self);
+                if(compute_configuration_file(self)) {
+                    if (self->configuration_scan.type == TYPE_LOCALSCAN) {
+                        //Launch localScan
+                        configure_local_scan(self);
+
+                        if (self->scan_size > 0) {
+                            zmsg_t *zmfalse = zmsg_new();
+                            zmsg_addstr(zmfalse, self->localscan_subscan.back().c_str());
+                            self->range_scan_config.range = zmsg_popstr(zmfalse);
+
+                            zsys_debug("Range scanner requested for %s with config file %s",
+                                    self->range_scan_config.range,
+                                    self->range_scan_config.config);
+
+                            // create range scanner
+                            if (self->range_scanner) {
+                                zpoller_remove(poller, self->range_scanner);
+                                zactor_destroy(&self->range_scanner);
+                            }
+                            reset_nb_discovered(self);
+                            self->range_scanner = zactor_new(range_scan_actor, &self->range_scan_config);
+                            zpoller_add(poller, self->range_scanner);
+                            self->status_scan = STATUS_PROGESS;
+
+                            zmsg_destroy(&zmfalse);
+                            zmsg_addstr(reply, RESP_OK);
+                        } else
+                            zmsg_addstr(reply, RESP_ERR);
+
+                    } else if ((self->configuration_scan.type == TYPE_MULTISCAN) ||
+                            (self->configuration_scan.type == TYPE_IPSCAN)) {
+                        //Launch rangeScan
+                        self->localscan_subscan = self->configuration_scan.scan_list;
+                        self->scan_size = self->configuration_scan.scan_size;
 
-                    if (self->scan_size > 0) {
                         zmsg_t *zmfalse = zmsg_new();
-                        zmsg_addstr(zmfalse, self->localscan_subscan.back().c_str());
-                        self->range_scan_config.range = zmsg_popstr(zmfalse);
+                        std::string next_scan = self->localscan_subscan.back();
+
+                        int ms_range_pos = next_scan.find("-");
+                        if (ms_range_pos == -1) {
+                            //Subnet
+                            zmsg_addstr(zmfalse, next_scan.c_str());
+                            self->range_scan_config.range = zmsg_popstr(zmfalse);
```

Comment:
this 2 step code is useless, reconsider it like this (more straightforward) :
 self->range_scan_config.range = strdup (next_scan.c_str())